Entitas/Entitas/CodeGenerator/Generators/PoolEntityGenerator.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Entitas.CodeGenerator;
+using Entitas.Serialization;
+
+namespace Entitas.CodeGenerator {
+
+    /// <summary>
+    /// Generates interfaces that declare what kind of components can be
+    /// created from a given pool.
+    /// </summary>
+    public class PoolEntityGenerator : IPoolEntityCodeGenerator {
+
+        const string FILE_NAME = "I{0}PoolEntity";
+        const string FILE_TEMPLATE = @"
+using Entitas;
+
+namespace Entitas {{
+
+    public interface {0} : IEntity {{{1}
+    }}
+}}
+";
+
+        public CodeGenFile[] Generate(ComponentInfo[] componentInfos) {
+
+            // Each component can have multiple pools. We want to group the
+            // components by their pools which means some components may show
+            // up in multiple groups. Here's what happens below:
+            // 
+            // 1. Filter the components that don't have pools.
+            // 2. We go through each component and expand based on the pools field.
+            //    e.g. a component with two pools becomes two components with one
+            //    pool each.
+
+            // 3. Now that we have a list of components with a single pool, we
+            //    perform a simple GroupBy().
+            // 4. For each group we create an interface. 
+
+            return componentInfos
+                .Where(component => component.pools.Length > 0)
+                .Aggregate(new List<ComponentInfo>(), (acc, component) => {
+                    var components = component.pools.Select(poolName => {
+                        return new ComponentInfo(
+                            component.fullTypeName,
+                            component.memberInfos,
+                            new[] { poolName },
+                            component.isSingleEntity,
+                            component.singleComponentPrefix,
+                            component.generateComponent,
+                            component.generateMethods,
+                            component.generateIndex
+                        );
+                    });
+                    acc.AddRange(components);
+                    return acc;
+                })
+                .GroupBy(component => component.pools[0])
+                .Where(group => group.Key.Length > 0)
```

Comment:
Just saw this :) https://github.com/sschmid/Entitas-CSharp/blob/d7fd022597516df36194ffa02fc77c77f37d6f51/Entitas/Entitas/CodeGenerator/Attributes/PoolAttribute.cs#L9